resources/sugarcrm/sugarjob.cpp

```diff
@@ -122,20 +123,17 @@ void SugarJob::Private::loginDone(const KDSoapGenerated::TNS__Set_entry_result &
     q->emitResult();
 }
 
-void SugarJob::Private::loginError(const KDSoapMessage &fault)
+void SugarJob::Private::loginError(int faultcode, const QString &messageError)
 {
     mSession->setSessionId(QString());
-
-    const int faultcode = fault.childValues().child(QLatin1String("faultcode")).value().toInt();
     kDebug() << q << "faultcode=" << faultcode;
     if (faultcode == QNetworkReply::UnknownNetworkError ||
```

Comment:
This code is wrong. The login() method takes care of QNetworkReply errors, and returns a job error code.
So loginError() should take an error code as input (not "faultcode") and just call q->setError() with it.